cmd/soroban-cli/src/commands/tx/new/revoke_sponsorship.rs

```diff
@@ -0,0 +1,139 @@
+                account_id: account_id_key,
+                signer_key: signer_key.clone(),
+            })
+        } else if let Some(asset) = &cmd.op.asset {
+            // Trustline sponsorship
+            let resolved_asset = cmd.tx.resolve_asset(asset)?;
+            let trustline_asset = match resolved_asset {
+                xdr::Asset::CreditAlphanum4(asset) => xdr::TrustLineAsset::CreditAlphanum4(asset),
+                xdr::Asset::CreditAlphanum12(asset) => xdr::TrustLineAsset::CreditAlphanum12(asset),
+                xdr::Asset::Native => xdr::TrustLineAsset::Native,
+            };
+            let ledger_key = xdr::LedgerKey::Trustline(xdr::LedgerKeyTrustLine {
+                account_id: account_id_key,
+                asset: trustline_asset,
+            });
+            xdr::RevokeSponsorshipOp::LedgerEntry(ledger_key)
+        } else if let Some(data_name) = &cmd.op.data_name {
+            // Data entry sponsorship
+            let data_name_xdr: xdr::StringM<64> =
+                data_name.parse().map_err(|_| tx::args::Error::InvalidHex {
+                    name: "data_name".to_string(),
+                    hex: "invalid data name".to_string(),
+                })?;
+            let ledger_key = xdr::LedgerKey::Data(xdr::LedgerKeyData {
+                account_id: account_id_key,
+                data_name: data_name_xdr.into(),
+            });
+            xdr::RevokeSponsorshipOp::LedgerEntry(ledger_key)
+        } else if let Some(offer_id) = cmd.op.offer_id {
+            // Offer sponsorship
+            let ledger_key = xdr::LedgerKey::Offer(xdr::LedgerKeyOffer {
+                seller_id: account_id_key,
+                offer_id: offer_id
+                    .try_into()
+                    .map_err(|_| tx::args::Error::InvalidHex {
+                        name: "offer_id".to_string(),
+                        hex: "offer ID too large".to_string(),
+                    })?,
+            });
+            xdr::RevokeSponsorshipOp::LedgerEntry(ledger_key)
+        } else if let Some(claimable_balance_id) = &cmd.op.claimable_balance_id {
+            // Claimable balance sponsorship
+            let balance_id_bytes = parse_balance_id(claimable_balance_id)?;
+            let mut balance_id_array = [0u8; 32];
+            balance_id_array.copy_from_slice(&balance_id_bytes);
+            let claimable_balance_id_xdr =
+                xdr::ClaimableBalanceId::ClaimableBalanceIdTypeV0(xdr::Hash(balance_id_array));
+            let ledger_key = xdr::LedgerKey::ClaimableBalance(xdr::LedgerKeyClaimableBalance {
+                balance_id: claimable_balance_id_xdr,
+            });
+            xdr::RevokeSponsorshipOp::LedgerEntry(ledger_key)
+        } else if let Some(liquidity_pool_id) = &cmd.op.liquidity_pool_id {
+            // Liquidity pool sponsorship
+            let pool_id_bytes =
+                hex::decode(liquidity_pool_id).map_err(|_| tx::args::Error::InvalidHex {
```

Comment:
Just like how claimable balances now have a strkey, liquidity pool's also have a strkey and we should support that here too.

Reply:
Good call. Done!